File: backend/internal/models/produto_mercearia.go
package models

type ProdutoMercearia struct {
	ID                  int           `json:"id"`
	LojaID              int           `json:"loja_id"`
	ProdutoGenericoID   int           `json:"produto_generico_id"`
	ProdutoGenericoNome string        `json:"produto_generico_nome,omitempty"`
	SKU                 string        `json:"sku"`
	Marca               string        `json:"marca"`
	Descricao           string        `json:"descricao"`
	CodigoBarras        string        `json:"codigo_barras"`
	QuantidadeEmbalagem float64       `json:"quantidade_embalagem"`
	UnidadeMedida       UnidadeMedida `json:"unidade_medida"`
	Ativo               bool          `json:"ativo"`
}

File: frontend/src/types/ProdutoMercearia.ts
export interface ProdutoMercearia {
  id?: number;
  produto_generico_id: number;
  produto_generico_nome?: string;
  sku: string;
  marca: string;
  descricao: string;
  codigo_barras: string;
  quantidade_embalagem: number;
  unidade_medida: string;
  ativo?: boolean;
}
